pylearn2/datasets/dataset.py

```diff
@@ -2,21 +2,25 @@
 A module defining the Dataset class.
 """
 
+import abc
+import warnings
+from pylearn2.utils.iteration import resolve_iterator_class
+
 
 class Dataset(object):
 
     """
     Abstract interface for Datasets.
     """
+    __metaclass__ = abc.ABCMeta
```

Comment:
Yeah, I was looking at Travis errors right now.
That line is not a problem, the @abstractmethod decorator of iterator though doesn't let you wrap the method from other classes. The test were failing cause DenseDesignMatrix.iterator wrapped Dataset.iterator. I just submitted a commit to fix it, let's see if it will pass the tests.